File: Java/src/main/java/parrot/AfricanParrot.java
package parrot;

/**
 * Created by damiengavard on 10/04/2017.
 */
public class AfricanParrot extends Parrot {

    private static final double LOAD_FACTOR = 9.0;
    private final int numberOfCoconuts;

    AfricanParrot(int numberOfCoconuts) {
        this.numberOfCoconuts = numberOfCoconuts;
    }

    @Override
    public double getSpeed() {
        return Math.max(0, getBaseSpeed() - LOAD_FACTOR * numberOfCoconuts);
    }

}

File: Java/src/main/java/parrot/AfricanParrot.kt
package parrot

class AfricanParrot internal constructor(private val numberOfCoconuts: Int) : Parrot() {

    companion object {
        private val LOAD_FACTOR = 9.0
    }

    override val speed get() = Math.max(0.0, baseSpeed - LOAD_FACTOR * numberOfCoconuts)
}